### PR TITLE
Fix slow annotations

### PR DESCRIPTION
**Pull request by Jipje**
Visualises annotations a bit more with colour and height. Very basic for debugging purposes. So is another PR for later. What it is now is really really fast. QUANTUM SHIT BABY

### CODE REVIEW

**Comment by Jipje** [comment]
Besides that also handles loading annotations without genomes.
It is so fast because of the implementation of a bucket list to hold the annotations.

File: src/main/java/gui/GraphDrawer.java [hunk 1]
```diff
@@ -223,65 +218,102 @@ private void drawNodes() {
...
-                    annoWidth = (annoWidth * (1 - (emptyAtStart / node.getSequenceLength())));
-                    startXAnno = startXAnno + (coordinates[2] - annoWidth);
-                }
-                if (endOfAnno < (node.getOffsets()[startCorOfGenome] + node.getSequenceLength())) {
-                    int emptyAtEnd = node.getOffsets()[startCorOfGenome] + node.getSequenceLength() - endOfAnno;
-                    annoWidth = (annoWidth * (1 - (emptyAtEnd / (node.getSequenceLength() - emptyAtStart))));
+            double annoHeight = coordinates[3] / 2;
+            double startYAnno = coordinates[1] + coordinates[3] - annoHeight;
+
+            for (Annotation annotation : drawThese) {
```
Comment (BugOrFeature) [inline src/main/java/gui/GraphDrawer.java]: Its way quicker now, well done.

File: src/main/java/gui/sub_controllers/ColourController.java [hunk 2]
```diff
@@ -160,6 +159,13 @@ public Color getBase() {
         return res;
     }
 
+    public Color getAnnotationColor(int startCorAnno, int stepSize) {
```
Comment (BugOrFeature) [inline src/main/java/gui/sub_controllers/ColourController.java]: nice color wheel implementation

File: src/main/java/structures/BinaryTree.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
+package structures;
+
+import java.util.Vector;
```
Comment (BugOrFeature) [inline src/main/java/structures/BinaryTree.java]: not used.

File: src/main/java/structures/TreeNode.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
+package structures;
+
+import java.util.ArrayList;
+
```
Comment (BugOrFeature) [inline src/main/java/structures/TreeNode.java]: not used.

File: src/main/java/structures/BinaryTree.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+/**
+ * Created by Jip on 19-6-2017.
+ */
+public class BinaryTree {
```
Comment (JaspervanTilburg) [inline src/main/java/structures/BinaryTree.java]: Good thinking using a tree

**Pull request merged**